nestjs/src/courses/mentors/mentors.controller.ts

```diff
@@ -48,11 +49,27 @@ export class MentorsController {
     @Param('mentorId', ParseIntPipe) mentorId: number,
     @Param('courseId', ParseIntPipe) courseId: number,
     @Req() req: CurrentRequest,
-  ) {
+  ): Promise<MentorDashboardDto[]> {
     const hasAccess = await this.mentorsService.canAccessMentor(req.user, mentorId);
     if (!hasAccess) {
       throw new ForbiddenException();
     }
     return await this.mentorsService.getStudentsTasks(mentorId, courseId);
   }
+
+  @Get('/:mentorId/course/:courseId/random-task')
+  @ApiOperation({ operationId: 'getRandomTask' })
+  @ApiOkResponse({ type: InsertResult })
+  @ApiBadRequestResponse()
+  public async getRandomTask(
+    @Param('mentorId', ParseIntPipe) mentorId: number,
+    @Param('courseId', ParseIntPipe) courseId: number,
+    @Req() req: CurrentRequest,
+  ): Promise<InsertResult> {
+    const hasAccess = await this.mentorsService.canAccessMentor(req.user, mentorId);
+    if (!hasAccess) {
+      throw new ForbiddenException();
+    }
+    return await this.mentorsService.getRandomTask(mentorId, courseId);
```

Comment:
consider adding proper dto rathen than exposing raw response of from typeorm

Reply:
Sure. I'm planning to do some refactoring here in next PR